beacon/beacon.go

```diff
@@ -262,9 +266,26 @@ func (pd *ProtocolDriver) Close() {
 	if err := pd.eg.Wait(); err != nil {
 		pd.logger.With().Info("received error waiting for goroutines to finish", log.Err(err))
 	}
+	close(pd.results)
 	pd.logger.Info("beacon goroutines finished")
 }
 
+func (pd *ProtocolDriver) onResult(epoch types.EpochID, beacon types.Beacon) {
+	select {
+	case pd.results <- result.Beacon{Epoch: epoch, Beacon: beacon}:
+	default:
+		pd.logger.With().Error("results queue is congested",
+			log.Uint32("epoch_id", epoch.Uint32()),
+			log.Stringer("beacon", beacon),
+		)
+	}
+}
+
+// Results notifies waiter that beacon for a target epoch has completed.
+func (pd *ProtocolDriver) Results() <-chan result.Beacon {
```

Comment:
i imagined that such notifications will be done in same place, for example

```
func watchBeacons(results <-chan Beacon) error {
     for beacon := results {
           tortoise.OnBeacon(beacon)
           builder.OnBeacon(beacon)
           hare.OnBeacon(beacon)
           beacons.Add(beacon) // persist
     }
}
```

maybe once every 6 month there will be new component that needs to watch beacon, so it will be just added to this loop, but this is still very easy to digest